addons/mass_mailing_sale/models/mailing_mailing.py

```diff
@@ -10,52 +10,71 @@ class MassMailing(models.Model):
     _name = 'mailing.mailing'
     _inherit = 'mailing.mailing'
 
-    sale_quotation_count = fields.Integer('Quotation Count', groups='sales_team.group_sale_salesman', compute='_compute_sale_quotation_count')
-    sale_invoiced_amount = fields.Integer('Invoiced Amount', groups='sales_team.group_sale_salesman', compute='_compute_sale_invoiced_amount')
+    sale_quotation_count = fields.Integer('Quotation Count', compute='_compute_sale_quotation_count')
+    sale_invoiced_amount = fields.Integer('Invoiced Amount', compute='_compute_sale_invoiced_amount')
 
     @api.depends('mailing_domain')
     def _compute_sale_quotation_count(self):
-        has_so_access = self.env['sale.order'].check_access_rights('read', raise_exception=False)
-        if not has_so_access:
-            self.sale_quotation_count = 0
-            return
         for mass_mailing in self:
-            mass_mailing.sale_quotation_count = self.env['sale.order'].search_count(mass_mailing._get_sale_utm_domain())
+            mass_mailing.sale_quotation_count = self.env['sale.order'].sudo().search_count(
```

Comment:
Now that sale orders are fetched using a simple domain, can't you do a ``read_group`` instead ? It would allow to have the counters directly with a single query.

Reply:
Like this?
```py
def _compute_sale_quotation_count(self):
    quotation_data = self.env['sale.order'].sudo().read_group(
        [('source_id', 'in', self.source_id.ids)],
        ['source_id'], ['source_id'],
    )
    mapped_data = {quotation['source_id'][0]: quotation['source_id_count'] for quotation in quotation_data}
    for mass_mailing in self:
        mass_mailing.sale_quotation_count = mapped_data.get(mass_mailing.source_id.id, 0)
```

Reply:
Seems right indeed